weather/module.py

```diff
@@ -1,23 +1,155 @@
+        self, json: dict, ctx: commands.Context, lang_preference: str
+    ) -> list[dict]:
+        """
+        example json: https://wttr.in/praha?lang=sk&format=j1
+        get useful data from json as list of individual days
+        """
+
+        # get individual days to extract data
+        weather = []
+        nearest_place = json["nearest_area"][0]["areaName"][0]["value"]
+        lang_preference = f"lang_{lang_preference}"
+        for i in range(NUM_OF_FORECAST_DAYS):
+            day = json["weather"][i]
+            day_dict = {
+                "date": day["date"],
+                "nearest_place": nearest_place,
+            }
+            day = day["hourly"]
+            for day_phase, hour in DAY_PHASES.items():
+                if lang_preference != "lang_en":
+                    day_dict.update(
+                        {
+                            _(ctx, f"{day_phase}"): {
+                                "state": day[hour][lang_preference][0]["value"],
+                                "temp": day[hour]["tempC"],
+                                "feels_like": day[hour]["FeelsLikeC"],
+                                "wind_speed": day[hour]["windspeedKmph"],
+                                "rain_chance": day[hour]["chanceofrain"],
+                            }
+                        }
+                    )
+                else:
+                    day_dict.update(
+                        {
+                            _(ctx, f"{day_phase}"): {
+                                "state": day[hour]["weatherDesc"][0]["value"],
+                                "temp": day[hour]["tempC"],
+                                "feels_like": day[hour]["FeelsLikeC"],
+                                "wind_speed": day[hour]["windspeedKmph"],
+                                "rain_chance": day[hour]["chanceofrain"],
+                            }
+                        }
+                    )
+
+            weather.append(day_dict)
+        return weather
+
+    async def _create_embeds(
+        self, ctx: commands.Context, name: str, lang_preference: str
+    ) -> list[nextcord.Embed]:
+        """create embeds for scrollable embed"""
+        url = f"https://wttr.in/{name}?format=j1&lang={lang_preference}"
+        try:
+            async with aiohttp.ClientSession() as session:
+                async with session.get(url) as resp:
+                    resp_json = await resp.json()
+        except aiohttp.ClientResponseError as e:
+            await guild_log.warning(
+                ctx.author,
+                ctx.channel,
+                f'An error occured while getting weather info, err code "{e.code}"',
+            )
+            return [
+                utils.discord.create_embed(
+                    author=ctx.message.author,
+                    title=_(ctx, "An error occured while getting weather info."),
+                    error=True,
+                )
+            ]
+        else:
```

Comment:
Because you are `return`ing in the `except` block, you can continue without wrapping the following code in `else`, am I right?

Reply:
Yes, you are right.